### PR TITLE
Change System ID and Component ID defaults and update logic

### PR DESCRIPTION
**Pull request by lbegani**
The changes are to update the system id and component id defaults and update logic to fix following issues - 
1. https://github.com/Dronecode/camera-streaming-daemon/issues/149
2. https://github.com/Dronecode/camera-streaming-daemon/issues/142

### CODE REVIEW

**Comment by lbegani** [comment]
@hamishwillee @shakthi-prashanth-m Kindly review

File: src/mavlink_server.cpp [hunk 1]
```diff
@@ -41,8 +41,9 @@ MavlinkServer::MavlinkServer(const ConfFile &conf, std::vector<std::unique_ptr<S
     , _is_running(false)
     , _timeout_handler(0)
     , _broadcast_addr{}
-    , _system_id(-1)
-    , _comp_id(MAV_COMP_ID_CAMERA2)
+    , _is_sys_id_found(false)
+    , _system_id(1)
```
Comment (shakthi-prashanth-m) [inline src/mavlink_server.cpp]: A `const` initializer for default system ID would be better.

File: src/mavlink_server.cpp [hunk 2]
```diff
@@ -41,8 +41,9 @@ MavlinkServer::MavlinkServer(const ConfFile &conf, std::vector<std::unique_ptr<S
...
+    , _comp_id(MAV_COMP_ID_CAMERA)
```
Comment (shakthi-prashanth-m) [inline src/mavlink_server.cpp]: Do you initialize this in cases with or without Gazebo ?

File: src/mavlink_server.cpp (hunk 1 above)
Comment (lbegani) [reply src/mavlink_server.cpp]: Fixed.

File: src/mavlink_server.cpp (hunk 2 above)
Comment (lbegani) [reply src/mavlink_server.cpp]: This is an usused variable, need to be removed later

File: src/mavlink_server.cpp [hunk 3]
```diff
@@ -68,20 +68,17 @@ MavlinkServer::MavlinkServer(const ConfFile &conf, std::vector<std::unique_ptr<S
...
-            log_error("Invalid Component ID for MAVLink communication (%d). Using default "
-                      "MAV_COMP_ID_CAMERA2 (%d)",
-                      opt.compid, MAV_COMP_ID_CAMERA2);
-        else
-            _comp_id = opt.compid;
+        } else {
+            log_error("Invalid System ID for MAVLink communication (%d)", opt.sysid);
+            log_info("Use System ID 1, till heartbeat received from Vehicle");
+            _system_id = 1;
+            _is_sys_id_found = false;
```
[the remark this answers is not in the document]
Comment (lbegani) [reply src/mavlink_server.cpp]: Fixed.

File: src/mavlink_server.cpp [hunk 4]
```diff
@@ -573,9 +570,12 @@ void MavlinkServer::_handle_heartbeat(const struct sockaddr_in &addr, mavlink_me
     mavlink_heartbeat_t heartbeat;
     mavlink_msg_heartbeat_decode(msg, &heartbeat);
 
-    if (heartbeat.autopilot == 12)
-        _system_id = msg->sysid;
-
+    if (heartbeat.autopilot == 12) {
```
[the remark this answers is not in the document]
Comment (lbegani) [reply src/mavlink_server.cpp]: Fixed.

File: src/mavlink_server.cpp [hunk 5]
```diff
@@ -778,14 +779,21 @@ int MavlinkServer::addCameraComponent(CameraComponent *camComp)
...
-            ret = id;
+    int compid;
+#ifdef ENABLE_GAZEBO
+    // PX4-SITL Gazebo is running a camera component with id MAV_COMP_ID_CAMERA
+    compid = MAV_COMP_ID_CAMERA2;
+#else
+    compid = MAV_COMP_ID_CAMERA;
+#endif
+
+    while (compid < MAV_COMP_ID_CAMERA6 + 1) {
```
[the remark this answers is not in the document]
Comment (lbegani) [reply src/mavlink_server.cpp]: Fixed.

File: src/mavlink_server.cpp [hunk 6]
```diff
@@ -75,10 +76,11 @@ MavlinkServer::MavlinkServer(const ConfFile &conf, std::vector<std::unique_ptr<S
 
         } else {
             log_error("Invalid System ID for MAVLink communication (%d)", opt.sysid);
-            log_info("Use System ID 1, till heartbeat received from Vehicle");
+            log_info("Use System ID %d, till heartbeat received from Vehicle", DEFAULT_SYSTEM_ID);
             _system_id = 1;
```
Comment (shakthi-prashanth-m) [inline src/mavlink_server.cpp]: Shouldn't this be 
```
_system_id = DEFAULT_SYSTEM_ID;
```

Comment (lbegani) [reply src/mavlink_server.cpp]: Fixed.

**Pull request merged**